src/NHibernate/Type/OneToOneType.cs

```diff
@@ -26,7 +26,7 @@ public override int GetColumnSpan(IMapping session)
 
 		public override SqlType[] SqlTypes(IMapping mapping)
 		{
-			return NoSqlTypes;
+			return GetIdentifierOrUniqueKeyType(mapping).SqlTypes(mapping);
```

Comment:
Is this change required? Hibernate did not change that.

Reply:
Required to make work queries with parameters `.Where(x => x.OneToOne = param)`.
Seems it's not fixed in Hibernate.